### ISSUE TITLE
Show a progress notification when a debug session is starting

### ISSUE THREAD

**Issue by jinyus** [Dart-Code/Dart-Code#4924]
**Describe the bug**
The status indicator in the gutter for test doesn't show the running state. Right now, you can only tell that the tests are running by locating the floating toolbar, but this gets confusing when you also have an emulator running which has its own toolbar.

**To Reproduce**
1. Create a test
2. Run it once
3. On subsequent runs, it doesn't show the running state, only `success` and `error`

**Expected behavior**
Display a loading spinner or another icon when a test in running. 

**Video**

https://github.com/Dart-Code/Dart-Code/assets/30532952/dfcbf346-4c8e-419c-bb14-891d86218473

 - Operating System and version: Linux 6.6.7-arch1-1 
 - VS Code version: 1.85.1
 - Dart extension version: v3.80.0
 - Dart/Flutter SDK version: Flutter 3.16.3 • channel stable | Dart 3.2.3 (stable)

**Comment by DanTup** [comment]
Can you show the whole source for your test? From the part shown here, it looks like the test is synchronous and would run very quickly, so it's possible that's why the spinner isn't showing. There's some time spent setting up the test run (spawning processes and loading the test before it begins executing) which won't be attributed to the test.

If I create a small sync test, I also don't see the running state, but if I slow the test down with a `Future.delayed()` then it's visible:


https://github.com/Dart-Code/Dart-Code/assets/1078012/0068f970-7c3b-49a5-a0c8-a3f3df1b4a00

**Comment by jinyus** [comment]
You're correct, I added the delay and it shows the spinner. Thanks!

**Comment by jinyus** [comment]
@DanTup How about showing the spinner as soon the run button is clicked rather than when the test actually starts? I think that would be a great UX improvement...what do you think?

**Comment by DanTup** [comment]
I think that could be difficult. We don't explicitly set the spinner, we basically just tell VS Code "this test is starting to run" (we call `TestRun.started(TestItem)` in the VS Code API). We currently do this when `pkg:test` tells us the test is starting, and the delay here is between you clicking "Run" and `pkg:test` telling us about the test.

It might seem like we could just call `started()` when we know that we're going to start running that test, but there are actually some edge cases here that mean the test you click "Run" on might not map 1-1 to an existing test we have in the tree - for example you might have:

```
for (var i in [1, 2]) {
  test('foo $i', () {
  });
}
```

So what happens here is we use the test we discovered _statically_ (which is called `foo $i`) to trigger the run, but when when `pkg:test` tells us it's starting tests, it tells us about `test 1` and `test 2` and we create new test items for those. It's those that we mark as started.

So, while we could try to mark the original statically-discovered test when you start a test run, I worry that it would be difficult to ensure we "end" all of the tests correctly (we'd need to complete the static test when the last real test we attribute to it completes).

But, I'll keep this open and have a pass through the code in case I can spot anything that might let us do this reliably.

**Comment by DanTup** [comment]
I had a look through this, but unfortunately I don't think there's a good way to do it. We don't get any event for the test until it starts running (and there may be other "tests" before it, such as those with the same name or `setUpAll` functions).

Marking the test as running would mean predicting the test node before we have the real name from `pkg:test` (which we can't guarantee to do reliably, which is why dynamic tests are handled with duplicate parent/child nodes, one that we created statically and one that was created at runtime). If we marked the parent static node as "running", we'd need to complete it with a result at the end, whereas the result is supposed to computed from the children (done automatically by VS Code).

Is the main issue here that you click to run a test, but get no visual feedback for a while? If so, I wonder if we should show a toast-notification indicating the debug session is starting (I thought we already did, but I don't see one today).
